Astro 3/3 Release Notes (#426)
astronomer/docs: astro/cli-release-notes.md

If you have any questions or a bug to report, don't hesitate to reach out to us via Slack or Intercom. We're here to help.

## v1.3.0

Release date: March 3, 2022

### New Command to Quickly Parse DAGs Before Deployment

> You can now use `astrocloud dev parse` to catch errors caused by missing OS Variables, Airflow Connections, and Airflow Variables in your DAGs,

Author: ```suggestion
You can now use `astrocloud dev parse` to run predefined collection of tests which ensure that essential aspects of your code are correctly formatted.
```

> You can now use `astrocloud dev parse` to catch errors caused by missing OS Variables, Airflow Connections, and Airflow Variables in your DAGs,

Author: Yeah this is better for release notes - wasn't sure when we were going to publish the other PRs, so wanted to get as much info here first and update it accordingly once we had more supporting docs

> When you run this command, the CLI runs several quick tests to ensure that your DAG code doesn't call any missing values. This includes calling all OS-level variables, Airflow Connections, and Airflow Variables to check whether each of those objects has a value. Additionally, this command runs prebuilt unit tests on your Astro project using Pytest. These unit tests are the same ones that are included by default with your project as described in [Test DAGs Locally with Pytest](test-and-troubleshoot-locally.md#test-dags-locally-with-pytest).

Author: ```suggestion
When you run this command, the CLI runs several quick tests to ensure that your DAG code doesn't call any missing values. Specifically, it calls all OS-level variables, Airflow Connections, and Airflow Variables to check whether each of those objects has a value. Additionally, this command runs prebuilt unit tests on your Astro project using Pytest. These unit tests are the same ones that are included by default with your project as described in [Test DAGs Locally with Pytest](test-and-troubleshoot-locally.md#test-dags-locally-with-pytest).
```

Reviewer: This needs to be updated see comments I left on other PR

> Generally speaking, `astrocloud dev parse` is a more robust but less customizable version of `astro dev pytest`. If don't have any specific test files that you want to run on your DAGs, then we recommend using `astrocloud dev parse` as your primary testing tool. 

Author: ```suggestion
Generally speaking, `astrocloud dev parse` is a more convenient but less customizable version of `astro dev pytest`. If don't have any specific test files that you want to run on your DAGs, then we recommend using `astrocloud dev parse` as your primary testing tool. 
```

### `astrocloud deploy` Parses DAGs by Default

:::danger Breaking Change

> `astrocloud deploy` no longer pushes code to your Deployment if the CLI detects basic errors in your DAGs. If any of your projects contain these errors, then certain deploys might stop working after you upgrade the CLI.

Reviewer: I think you need to say somewhere in here that the this will only happen if you are using runtime 4.1.0 and above

To maintain the CLI's original behavior, use `astrocloud deploy --force` to force a deploy even if your project might contain errors.

:::

> To better protect your Deployments from unexpected errors, `astrocloud deploy` now automatically applies tests from `astrocloud dev parse` to your Astro project before pushing new code to a Deployment. If any of these tests fail, the CLI will not push your code to Astro.

Reviewer: I would say it is just one test that checks for errors. The other "tests" are excluding variables and connections

> To better protect your Deployments from unexpected errors, `astrocloud deploy` now automatically applies tests from `astrocloud dev parse` to your Astro project before pushing new code to a Deployment. If any of these tests fail, the CLI will not push your code to Astro.

Author: I think we should still put focus on "original behavior" in any breaking change notes. The implication of the base note is that you need to start checking DAGs for errors, so don't think we need to repeat this advice in the warning note.

### New Command to Update Deployment Configurations

You can now use `astrocloud deployment update` to update a Deployment's core configuration from your CLI terminal. The configurations that you can update include:

- Deployment name
- Deployment description
- Scheduler resources
- Scheduler replicas
- Worker resources

> These are the same configurations that you can set via the **Edit Configuration** button in the Cloud UI. For more information about these configurations, see [Configure a Deployment](configure-deployment.md).

Author: I don't think we imply anywhere that you can update envars or API keys, so generally not a fan of saying what something "doesn't" do unless there's an alternative or additional consideration there.

If you have any questions or a bug to report, don't hesitate to reach out to us via Slack or Intercom. We're here to help.

## v1.3.0

Release date: March 3, 2022

### New Command to Parse DAGs for Errors

> `astrocloud dev parse` is a new Astro CLI command that allows you to run a set of basic tests against your Astro project to ensure that essential aspects of your code are properly formatted.

Reviewer: ```suggestion
`astrocloud dev parse` is a new Astro CLI command that allows you to run a basic test against your Astro project to ensure that essential aspects of your code are properly formatted.
```

> This includes the DAG integrity test that is run with `astro dev pytest`, as well as checks to ensure that your DAGs will run on the Airflow Webserver.

Reviewer: ```suggestion
This includes the DAG integrity test that is run with `astro dev pytest` which checks that your DAGs will run on the Airflow Webserver.
```

> When you run this command, the CLI runs several quick tests to ensure that your DAG code doesn't call any missing values. This includes calling all OS-level variables, Airflow Connections, and Airflow Variables to check whether each of those objects has a value. Additionally, this command runs prebuilt unit test using Pytest. This test is the same one that's included by default in your project as described in [Test DAGs Locally with Pytest](test-and-troubleshoot-locally.md#test-dags-locally-with-pytest).

Author: ```suggestion
When you run this command, the CLI runs several quick tests to ensure that your DAG code doesn't call any missing values. This includes calling all OS-level variables, Airflow Connections, and Airflow Variables to check whether each of those objects has a value. Additionally, this command runs prebuilt unit test using Pytest. These unit tests are the same ones that are included by default with your project as described in [Test DAGs Locally with Pytest](test-and-troubleshoot-locally.md#test-dags-locally-with-pytest).
```

> Generally speaking, `astrocloud dev parse` is a more robust but less customizable testing command than `astro dev pytest`. If don't have any specific test files that you want to run on your DAGs, then we recommend using `astrocloud dev parse` as your primary testing tool. 

Author: ```suggestion
Generally speaking, `astrocloud dev parse` is a more robust but less customizable version of `astro dev pytest`. If don't have any specific test files that you want to run on your DAGs, then we recommend using `astrocloud dev parse` as your primary testing tool. 
```

### `astrocloud deploy` Parses DAGs by Default

:::danger Breaking Change

`astrocloud deploy` no longer pushes code to your Deployment if the CLI detects basic errors in your DAGs. If any of your projects contain these errors, then certain deploys might stop working after you upgrade the CLI.

To maintain the CLI's original behavior, use `astrocloud deploy --force` to force a deploy even if your project might contain errors.

:::

To better protect your Deployments from unexpected errors, `astrocloud deploy` now automatically applies tests from `astrocloud dev parse` to your Astro project before pushing new code to a Deployment. If any of these tests fail, the CLI will not push your code to Astro.

### New Command to Update Deployment Configurations

You can now use `astrocloud deployment update` to update a Deployment's core configuration from your CLI terminal. The configurations that you can update include:

- Deployment name
- Deployment description
- Scheduler resources
- Scheduler replicas
- Worker resources

> These are the same configurations that you can set via the **Edit Configuration** page in the Cloud UI. For more information about these configurations, see [Configure a Deployment](configure-deployment.md).

Author: ```suggestion
These are the same configurations that you can set via the **Edit Configuration** button in the Cloud UI. For more information about these configurations, see [Configure a Deployment](configure-deployment.md).
```
